blockchain/metrics.go

```diff
@@ -0,0 +1,81 @@
+package blockchain
+
+import (
+	"github.com/go-kit/kit/metrics"
+	"github.com/go-kit/kit/metrics/discard"
+	prometheus "github.com/go-kit/kit/metrics/prometheus"
+	stdprometheus "github.com/prometheus/client_golang/prometheus"
+)
+
+// Metrics represents the blockchain metrics
+type Metrics struct {
+	// Gas Price Average
+	GasPriceAvg metrics.Histogram
+	// Gas used
+	GasUsed metrics.Histogram
+	// Block height
+	BlockHeight metrics.Gauge
+	// Block write duration time
+	BlockWriteDuration metrics.Histogram
+	// Transaction number
+	TransactionNum metrics.Histogram
+}
+
+// GetPrometheusMetrics return the blockchain metrics instance
+func GetPrometheusMetrics(namespace string, labelsWithValues ...string) *Metrics {
+	labels := []string{}
+
+	for i := 0; i < len(labelsWithValues); i += 2 {
+		labels = append(labels, labelsWithValues[i])
+	}
+
+	return &Metrics{
```

Comment:
Provide some reasonable buckets instead of default ones.
Otherwise, the histogram don't make much sense.